createWebNavigator.js

```diff
@@ -1,39 +1,38 @@
 import _ from 'underscore';
 import * as React from 'react';
-import {View, Pressable} from 'react-native';
+import {View, Dimensions} from 'react-native';
 import {useNavigationBuilder, createNavigatorFactory, StackRouter} from '@react-navigation/native';
+import RHPContainer from './components/RHPContainer';
+
+const isSmallScreenWidth = Dimensions.get('window').width <= 800;
+
 
 function CustomStackNavigator(props) {
     const {navigation, state, descriptors, NavigationContent} = useNavigationBuilder(StackRouter, {
       children: props.children,
       screenOptions: props.screenOptions,
       initialRouteName: props.initialRouteName,
     });
-
+    const isLHNVisible = (state.index === 0 && state.routes.length === 1) || !isSmallScreenWidth;
     const lastChatIndex = _.findLastIndex(state.routes, {name: 'Chat'});
     const lastRHPIndex = _.findLastIndex(state.routes, (route) => route.name !== 'Chat' && route.name !== 'LeftHandNav');
     const isRHPOnTopOfStack = lastRHPIndex === state.index;
     return (
         <NavigationContent>
             <View style={{flexDirection: 'row', flex: 1}}>
                 {state.routes.map((route, i) => {
-                    if (route.name === 'LeftHandNav') {
+                    if (isLHNVisible && i === 0) {
                         return (
                             <View
                                 key={route.key}
-                                style={{maxWidth: 350, flex: 1, borderRightWidth: 1}}
+                                style={{maxWidth: !isSmallScreenWidth ? 350 : Dimensions.get('window').width, flex: 1, borderRightWidth: !isSmallScreenWidth ? 1 : 0}}
                             >
                                 {descriptors[route.key].render()}
                             </View>
                         )
                     }
 
-                    // Only show the top-most chat and hide any others
-                    if (route.name === 'Chat' && i !== lastChatIndex) {
-                        return null;
-                    }
-
-                    if (route.name === 'Chat') {
+                    if (i === lastChatIndex) {
```

Comment:
I don't really see how this incorporates the feedback to not use scree names (if that's why we decided to use an index). We are still using the name of the route to derive the "chat".

So I think this line here:

https://github.com/marcaaron/navigation-test/blob/8f144a15a5c5a9c40ed5f4ff6dc545870a362f30/createWebNavigator.js#L17

would change to something more generic like "main" instead of "chat" and we'd ideally not use the `name` field but something like `isCenterScreen` or `isMainScreen` and then avoid the dependency on names.

Let me know if that makes sense or if you discovered any issues with it.

Reply:
I am not entirely sure how that would be any different from using the screen name? But I looked at the feedback in other PR and feel like I could explore some more options here.

Reply:
Well again if someone changes the name of the screen you will have to change it in the custom navigator. I think it's just a good "design" practice to not have a dependency like this. But since we're the only ones using this code I agree it doesn't matter.